templates/todo/api/csharp-cosmos-sql/Program.cs

```diff
@@ -1,41 +1,38 @@
-using Azure.Identity;
-using Microsoft.ApplicationInsights.AspNetCore.Extensions;
-using Microsoft.Azure.Cosmos;
-using SimpleTodo.Api;
-
-var credential = new DefaultAzureCredential();
-var builder = WebApplication.CreateBuilder(args);
-
-builder.Services.AddSingleton<ListsRepository>();
-builder.Services.AddSingleton(_ => new CosmosClient(builder.Configuration["AZURE_COSMOS_ENDPOINT"], credential, new CosmosClientOptions()
-{
-    SerializerOptions = new CosmosSerializationOptions
-    {
-        PropertyNamingPolicy = CosmosPropertyNamingPolicy.CamelCase
-    }
-}));
-builder.Services.AddControllers();
-builder.Services.AddApplicationInsightsTelemetry(builder.Configuration);
-
-var app = builder.Build();
-
-app.UseCors(policy =>
-{
-    policy.AllowAnyOrigin();
-    policy.AllowAnyHeader();
-    policy.AllowAnyMethod();
-});
-
-// Swagger UI
-app.UseSwaggerUI(options => {
-    options.SwaggerEndpoint("./openapi.yaml", "v1");
-    options.RoutePrefix = "";
-});
-
-app.UseStaticFiles(new StaticFileOptions{
-    // Serve openapi.yaml file
-    ServeUnknownFileTypes = true,
-});
-
-app.MapControllers();
+using Azure.Identity;
+using Microsoft.Azure.Cosmos;
+using SimpleTodo.Api;
+
+var builder = WebApplication.CreateBuilder(args);
+var credential = new DefaultAzureCredential();
+builder.Configuration.AddAzureKeyVault(new Uri(builder.Configuration["AZURE_KEY_VAULT_ENDPOINT"]), credential);
+
+builder.Services.AddSingleton<ListsRepository>();
+builder.Services.AddSingleton(_ => new CosmosClient(builder.Configuration[builder.Configuration["AZURE_COSMOS_CONNECTION_STRING_KEY"]]));

```

Comment:
Why not using DefaultAzureCredential and endpoint?

Reply:
In all cases we want to favor RBAC over connection string usage.

Reply:
@jongio This already existed in the sample. More than happy to update but didn't want to update too much.

Reply:
If you don't want to take it in this PR, then let's keep as-is, use RBAC if already using RBAC, use KEY if using key, then please create an issue to get them all updated to use RBAC.

Reply:
Doing in another PR per @jongio